tools/cas/cas.go

```diff
@@ -100,8 +101,16 @@ func main() {
 	bbClient := bbspb.NewBuildBuddyServiceClient(conn)
 
 	ctx := context.Background()
-	if *apiKey != "" {
-		ctx = metadata.AppendToOutgoingContext(ctx, "x-buildbuddy-api-key", *apiKey)
+
+	bbApiKey := *apiKey
+	if bbApiKey == "" {
+		// If the API key is not set, try to get it from BuildBuddy section in Git config.
+		if key, err := storage.ReadRepoConfig("api-key"); err == nil {
+			bbApiKey = key
+		}
+	}
+	if bbApiKey != "" {
+		ctx = metadata.AppendToOutgoingContext(ctx, "x-buildbuddy-api-key", bbApiKey)
```

Comment:
I could see this approach making sense for `bb download` since it's part of the CLI, but we probably shouldn't have deps from `//tools` => `//cli`.

We probably should just redact any args from the BES that contain `api_key` in `redact.go`

Reply:
It's `//tools` depending on `//cli` here, not the reverse, which I think is ok? We do want a set of common libraries for our CLIs (tools, enterprise/tools, cli) in general right?

I could look into `redact.go` separately.

Reply:
I think we shouldn't have non-CLI packages depending on CLI-specific functionality. The "ReadRepoConfig" is a CLI-specific config right now; it's sort of "owned" by the CLI. So it's a bit weird to have non-CLI tools reading that data. This `//tools` directory is for our internal ad-hoc tooling and the shared libraries that we use for tools are all under //server today, not //cli.

This tool is sort of redundant with `bb download` and we want to remove it eventually. I think we want to deprecate this tool and switch to `bb download` for everything.

Reply:
SGTM